logdevice/ops/ldquery/tables/LogDirectories.cpp

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+#include "logdevice/ops/ldquery/tables/LogDirectories.h"
+
+#include <queue>
+#include <folly/Conv.h>
+#include <folly/json.h>
+
+#include "../Table.h"
+#include "../Utils.h"
+#include "LogTreeBase.h"
+
+#include "logdevice/common/configuration/Configuration.h"
+#include "logdevice/common/configuration/ReplicationProperty.h"
+#include "logdevice/common/configuration/UpdateableConfig.h"
+#include "logdevice/common/debug.h"
+#include "logdevice/lib/ClientImpl.h"
+
+
+namespace facebook {
+  namespace logdevice {
+    namespace ldquery {
+      namespace tables {
+
+TableColumns LogDirectories::getColumns() const {
+  return LogTreeBase::getAttributeColumns();
+}
+
+std::shared_ptr<TableData> LogDirectories::getData(QueryContext& ctx) {
+  auto result = std::make_shared<TableData>();
+  auto client = ld_ctx_->getFullClient();
+
+  ld_check(client);
+
+  auto add_row =
+      [&](client::Directory* dir, const std::string path) {
+    auto dir_attrs = dir->attrs();
+
+    // This should remain the first ColumnValue as expected by the code below.
+    result->cols["parent_dir"].push_back(path);
+    result->cols["name"].push_back(dir->getFullyQualifiedName());
+    result->cols["logid_lo"].push_back(std::string("0"));
+    result->cols["logid_hi"].push_back(std::string("0"));
+
+    LogTreeBase::populateAttributeData(*result, dir_attrs);
+  };
+
+  // If the query contains a constraint on the directory name, we can efficiently
+  // find the directory using getDirectorySync().
+  std::string expr;
+  if (columnHasEqualityConstraint(10, ctx, expr)) {
+      auto parent_dir = client->getDirectorySync(expr);
+      if (parent_dir) {
+          const client::DirectoryMap& children = parent_dir->children();
+          for (auto& dir_pair : children) {
+              add_row(dir_pair.second.get(), parent_dir->getFullyQualifiedName());
+          }
+      }
+  } else if (columnHasEqualityConstraint(0, ctx, expr)) {
+    auto dir = client->getDirectorySync(expr);
+    if (dir) {
+      add_row(dir.get(), dir->parent()->getFullyQualifiedName());
+    }
+  } else if (columnHasLikeConstraint(0, ctx, expr)) {
```

Comment:
You do not need to implement you own LIKE operator: when LDQuery processes your request it uses this function to fetch the data, then uses fetched data to populate SQLite table and then runs provided SQL query on this table. So in theory we do not need to do *any* filtering in this function and still will be able to use the full power of SQL to filter them.
However, in practice you often can run query in more efficient manner by pushing down the filters. Then instead of fetching all the data and filtering you can filter at the source and get only requested subset of data — this is, for example, why we have 2 code paths for fetching LogGroups one by name (efficient) and another for all (slower but we can apply complex filters later).
In you case pushing down LIKE is both complex and will not give us any additional benefits, so instead you can choose the same path as in LogGroup table: either fetch one item by name or all. Then you will be able to write queries like "SELECT * from log_dirs WHERE name LIKE '%my_dir%'" because all the heavy will be done by LDQuery itself.

Reply:
Busy working on something else recently, will update this PR once I got some time :-)